lib/common.sh

```diff
@@ -196,6 +196,11 @@ process_status(){
 #
 equals(){
   [[ "${1}" == "${2}" ]]; process_status "$?"
+  RET_CODE="$?"
```

Comment:
Since you're not using RESULT_STR, would it be cleaner to just not use process_status here and do something like:

```
equals(){
  if [[ "${1}" != "${2}" ]]; then
    echo "ERROR: expected ${2}, got ${1}"
    FAILS=$((FAILS+1))
  fi
}
```

Reply:
process_status is printing the RESULT_STR that is set before calling the equals function, so we either use process_status or we need to add RESULT_STR in the output.

Reply:
Ok I see now - it still looks a bit weird, because RET_CODE is saving the return code from process_status, not the original return-code from the test, would it be clearer to do something like:

```
equals(){
  [[ "${1}" != "${2}" ]]; RET_CODE="$?
  if ! process_status $RET_CODE; then
    echo "        expected ${2}, got ${1}"
  fi
}
```

Reply:
ok, I'll fix it. thank you